Apeswap-Bots/Address-Inclusion-Exclusion-Detection/src/agent.spec.ts

```diff
@@ -0,0 +1,165 @@
+        alertId: "APESWAP-2-1",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: "Apeswap",
+        metadata: {
+          "token Address": tokenInfo.address,
+          "excluded Address": account,
+        },
+      });
+    default:
+      return Finding.fromObject({
+        name: "New Address inclusion",
+        description: `${name} function call detected from ${tokenInfo.name} contract`,
+        alertId: "APESWAP-2-2",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: "Apeswap",
+        metadata: {
+          "token Address": tokenInfo.address,
+          "included Address": account,
+        },
+      });
+  }
+};
+
+const createFunctionInterface = (functionName: string, ...functionParameters: parameterDefinition[]): AbiItem => ({
+  type: "function",
+  name: functionName,
+  inputs: functionParameters.map(({ type, name }) => ({
+    type,
+    name,
+  })),
+});
+
+const createExclusionTransactionData = (addrToExcl: string): string =>
+  encodeFunctionCall(createFunctionInterface("excludeAccount", { name: "account", type: "address" }), [addrToExcl]);
+
+const createInclusionTransactionData = (addrToIncl: string): string =>
+  encodeFunctionCall(createFunctionInterface("includeAccount", { name: "account", type: "address" }), [addrToIncl]);
+
+const TOKEN_CONTRACT: testTokenType = {
+  address: createAddress("0x487GC"),
+  name: "Test token",
+};
+
+describe("address inclusion and exclusion bot test", () => {
+  const handleTransaction: HandleTransaction = provideBotHandler(TOKEN_CONTRACT);
+  const transferTransactionData = encodeFunctionCall(
+    createFunctionInterface("transfer", { name: "value", type: "uint256" }, { name: "to", type: "address" }),
+    ["100000000000000", createAddress("0x123")]
+  );
+
+  it("should return empty findings if there are no inclusion/exclusion function calls", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent().setData(transferTransactionData);
+    const findings: Finding[] = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it(`should return empty findings if there is an inclusion/exclusion function call not invoked from ${TOKEN_CONTRACT.name}`, async () => {
+    let txEvent: TransactionEvent = new TestTransactionEvent()
+      .setData(createExclusionTransactionData(createAddress("0x1237")))
+      .setTo(createAddress("0x568A"));
+    let findings: Finding[] = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+
+    txEvent = new TestTransactionEvent()
+      .setData(createInclusionTransactionData(createAddress("0x1389")))
+      .setTo(createAddress("0x597A"));
+    findings = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it(`should return a finding if there is an inclusion/exclusion function invocation from  ${TOKEN_CONTRACT.name}`, async () => {
```

Comment:
I prefer to remove the string interpolation from the name of the tests to improve readability.